key.go

```diff
@@ -570,11 +570,21 @@ func readInputs(ctx context.Context, msgs chan<- Msg, input io.Reader) error {
 
 var unknownCSIRe = regexp.MustCompile(`^\x1b\[[\x30-\x3f]*[\x20-\x2f]*[\x40-\x7e]`)
 
+var mouseSGRRegex = regexp.MustCompile(`(\d+);(\d+);(\d+)([Mm])`)
+
 func detectOneMsg(b []byte) (w int, msg Msg) {
 	// Detect mouse events.
+	// X10 mouse events have a length of 6 bytes
 	const mouseEventLen = 6
-	if len(b) >= mouseEventLen && b[0] == '\x1b' && b[1] == '[' && b[2] == 'M' {
-		return mouseEventLen, MouseMsg(parseX10MouseEvent(b))
+	if len(b) >= mouseEventLen && b[0] == '\x1b' && b[1] == '[' {
+		switch b[2] {
+		case 'M':
+			return mouseEventLen, MouseMsg(parseX10MouseEvent(b))
+		case '<':
+			if mouseSGRRegex.Match(b[3:]) {
+				return mouseEventLen, MouseMsg(parseSGRMouseEvent(b))
+			}
+		}
 	}
```

Comment:
You have to fix the mouseEventLen here otherwise it will throw after each detected sgr event random keyRunes Events with left over chars. Bad for focused input fields and so on. SGR events dont have like X10 Events a fixed length of 6. Otherwise very cool. Works very clean on windows!

My recommended solution:

```suggestion
func detectOneMsg(b []byte) (w int, msg Msg) {
	// Detect mouse events.
	// X10 mouse events have a length of 6 bytes
	const mouseEventX10Len = 6
	if len(b) >= mouseEventX10Len && b[0] == '\x1b' && b[1] == '[' {
		switch b[2] {
		case 'M':
			return mouseEventX10Len, MouseMsg(parseX10MouseEvent(b))
		case '<':
			if matchIndices := mouseSGRRegex.FindSubmatchIndex(b[3:]); matchIndices != nil {
				// SGR mouse events length is the length of the match plus the length of the escape sequence
				mouseEventSGRLen := matchIndices[1] + 3
				return mouseEventSGRLen, MouseMsg(parseSGRMouseEvent(b))
			}
		}
	}
 
```

Would love to see this soon be merged into main branch. 😄

Reply:
@robin-samuel good catch! It would be nice to have a test case for this 🙂

Reply:
```go
package main

import (
	"github.com/charmbracelet/bubbles/textinput"
	tea "github.com/charmbracelet/bubbletea"
	"github.com/charmbracelet/lipgloss"
)

type Model struct {
	width  int
	height int
	input  textinput.Model
}

func New() *Model {
	ti := textinput.New()
	ti.Placeholder = "Type something..."
	ti.Focus()
	return &Model{
		input: ti,
	}
}

func (m *Model) Init() tea.Cmd {
	return textinput.Blink
}

func (m *Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	cmds := make([]tea.Cmd, 0)
	switch msg := msg.(type) {
	case tea.WindowSizeMsg:
		m.width = msg.Width
		m.height = msg.Height
	case tea.KeyMsg:
		switch msg.Type {
		case tea.KeyCtrlC:
			return m, tea.Quit
		}
	}
	ti, cmd := m.input.Update(msg)
	m.input = ti
	if cmd != nil {
		cmds = append(cmds, cmd)
	}
	return m, tea.Batch(cmds...)
}

func (m *Model) View() string {
	style := lipgloss.NewStyle().
		Width(m.width).
		Height(m.height)
	return style.Render(lipgloss.JoinVertical(lipgloss.Left,
		"Move your mouse around over the terminal window.",
		"You will see random inputs appear in the text input field below.",
		m.input.View(),
		"Press Ctrl+C to quit.",
	))
}

func main() {
	p := tea.NewProgram(New(), tea.WithAltScreen(), tea.WithMouseAllMotion())
	if _, err := p.Run(); err != nil {
		panic(err)
	}
}
```

> @robin-samuel good catch! It would be nice to have a test case for this 🙂

Here a very simple example. By moving the mous over the window you will see that the input gets spammed with key.Runes events and this text will automatically be put in the input field. With the fix applied this wont be the case and act like usual! 🙂

Reply:
@robin-samuel thank you 🙂